notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1136_M1M3_analysis_stability_tracking.ipynb

```diff
@@ -0,0 +1,704 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #8.    print(f'Found {len(slews)} slews and {len(tracks)} tracks')
</pre><p> </p><p>When I try to run this cell on my "local" usdf notebook , I get the following error:</p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[20], line 3
      1 # Select data from a given date
      2 eventMaker = TMAEventMaker()
----&gt; 3 events = eventMaker.getEvents(dayObs)
      5 # Get lists of slew and track events
      6 slews = [e for e in events if e.type==TMAState.SLEWING]

File ~/notebooks/lsst-sitcom/summit_utils/python/lsst/summit/utils/tmaUtils.py:1096, in TMAEventMaker.getEvents(self, dayObs)
   1092     return []
   1094 # applies the data to the state machine, and generates events from the
   1095 # series of states which results
-&gt; 1096 events = self._calculateEventsFromMergedData(data, dayObs, dataIsForCurrentDay=workingLive)
   1097 if not events:
   1098     self.log.warning(f"Failed to calculate any events for {dayObs=} despite EFD data existing!")

File ~/notebooks/lsst-sitcom/summit_utils/python/lsst/summit/utils/tmaUtils.py:1205, in TMAEventMaker._calculateEventsFromMergedData(self, data, dayObs, dataIsForCurrentDay)
   1203 stateTuples = self._statesToEventTuples(tmaStates, dataIsForCurrentDay)
   1204 events = self._makeEventsFromStateTuples(stateTuples, dayObs, data)
-&gt; 1205 self.addBlockDataToEvents(dayObs, events)
   1206 return events

File ~/notebooks/lsst-sitcom/summit_utils/python/lsst/summit/utils/tmaUtils.py:1337, in TMAEventMaker.addBlockDataToEvents(self, dayObs, events)
   1335 for block, seqNums in blockDict.items():
   1336     for seqNum in seqNums:
-&gt; 1337         blockInfo = blockParser.getBlockInfo(block=block, seqNum=seqNum)
   1339         relatedEvents = blockParser.getEventsForBlock(events, block=block, seqNum=seqNum)
   1340         for event in relatedEvents:

File ~/notebooks/lsst-sitcom/summit_utils/python/lsst/summit/utils/blockUtils.py:414, in BlockParser.getBlockInfo(self, block, seqNum)
    412 # likewise for the blockIds
    413 if len(blockIds) &gt; 1:
--&gt; 414     raise RuntimeError(f"Found multiple blockIds ({blockIds}) for {seqNum=}")
    415 blockId = blockIds.pop()
    417 blockInfo = BlockInfo(
    418     blockNumber=block,
    419     blockId=blockId,
   (...)
    426     states=statePoints,
    427 )

RuntimeError: Found multiple blockIds ({'BL13_O_20231219_000001', 'BL13_O_20231220_000001'}) for seqNum=1
</pre><p> </p><p>We need to make sure that the notebook works when other people try to execute it. If further action is required from user, add a comment guiding user on how to proceed. </p><p> </p><p>I was planning to do some tests locally (for example, on the rms definition or on how to you calculated the indexes), but could not do due to this failure.</p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/107/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
<p>This bug also occurred to me. Someone commented in slack what to do to fix it. I'll look it up and include it as a comment.</p>

Reply:
<p>I have just added a small change in place of a comment. Let me know if it works or not yet.</p>

Reply:
<p>Sorry, kind of my mistake, since I was not using the latest versions of summit_utils package. But anyways it is very useful to have your comment in there.</p>